paper-api-generator/src/main/java/io/papermc/generator/types/goal/MobGoalGenerator.java

```diff
@@ -0,0 +1,260 @@
+        new DeprecatedEntry(Tameable.class, "owner_hurt_by_target", "1.21", null),
+        new DeprecatedEntry(Tameable.class, "owner_hurt_target", "1.21", null),
+        new DeprecatedEntry(Parrot.class, "perch", "1.21", null),
+        new DeprecatedEntry(Raider.class, "raid", "1.21", null),
+        new DeprecatedEntry(Creature.class, "random_fly", "1.21", null),
+        new DeprecatedEntry(Mob.class, "random_lookaround", "1.21", null),
+        new DeprecatedEntry(Creature.class, "random_stroll_land", "1.21", null),
+        new DeprecatedEntry(Creature.class, "random_swim", "1.21", null),
+        new DeprecatedEntry(Tameable.class, "random_target_non_tamed", "1.21", null),
+        new DeprecatedEntry(Tameable.class, "sit", "1.21", null),
+        new DeprecatedEntry(Creature.class, "stroll_village", "1.21", null),
+        new DeprecatedEntry(AbstractHorse.class, "tame", "1.21", null),
+        new DeprecatedEntry(Creature.class, "water", "1.21", null),
+        new DeprecatedEntry(Dolphin.class, "water_jump", "1.21", null),
+        new DeprecatedEntry(Creature.class, "stroll_village_golem", "1.21", null),
+        new DeprecatedEntry(Mob.class, "universal_anger_reset", "1.21", null)
+        //</editor-fold>
+    };
+
+    public MobGoalGenerator(final String keysClassName, final String pkg) {
+        super(keysClassName, pkg);
+    }
+
+    @Override
+    protected TypeSpec getTypeSpec() {
+        final TypeVariableName tExtendsMobGeneric = TypeVariableName.get("T", Mob.class);
+        final TypeName clazzType = notNullParamType(Class.class, tExtendsMobGeneric);
+        final TypeName keyType = notNullType(String.class);
+        final MethodSpec.Builder createMethod = MethodSpec.methodBuilder("create")
+            .addTypeVariable(tExtendsMobGeneric)
+            .addModifiers(PRIVATE, STATIC)
+            .addParameter(keyType, "key", FINAL)
+            .addParameter(clazzType, "clazz", FINAL)
+            .addCode("return $T.of(clazz, $T.minecraft(key));", GoalKey.class, NamespacedKey.class)
+            .returns(notNullParamType(GoalKey.class, tExtendsMobGeneric));
+
+
+        final TypeSpec.Builder typeBuilder = TypeSpec.interfaceBuilder(this.className)
+            .addSuperinterface(ParameterizedTypeName.get(ClassName.get(com.destroystokyo.paper.entity.ai.Goal.class), tExtendsMobGeneric))
+            .addModifiers(PUBLIC)
+            .addTypeVariable(tExtendsMobGeneric)
+            .addAnnotations(Annotations.CLASS_HEADER)
+            .addJavadoc(CLASS_HEADER);
+
+
+        final Stream<Class<? extends Goal>> classes;
+        try (final ScanResult scanResult = new ClassGraph().enableAllInfo().whitelistPackages("net.minecraft").scan()) {
+            classes = scanResult.getSubclasses(Goal.class.getName()).loadClasses().stream().map(c -> c.asSubclass(Goal.class));
+        }
+
+        final List<VanillaGoalKey> vanillaNames = classes
+            .filter(MobGoalGenerator::hasNoEnclosingClass)
```

Comment:
It's worth trying to remove this filter and see what happens.
I don't fully recall, but I think I filtered out inner classes because they weren't named on spigot mappings so we couldn't generate a proper key, that limitation isn't a thing anymore so we should be able to generate more keys

Reply:
Removed